components/expression/__examples__/example.jsx

```diff
@@ -0,0 +1,498 @@
+];
+
+const OperatorsList = [
+	{ id: '1', label: 'Equals' },
+	{ id: '2', label: 'Does Not Equals' },
+	{ id: '3', label: 'Greater Than' },
+	{ id: '4', label: 'Less Than' },
+];
+
+class Example extends React.Component {
+	static displayName = 'ExpressionBaseExample';
+
+	constructor(props) {
+		super(props);
+		this.state = {
+			conditions: props.conditions,
+			triggerType: props.triggerType,
+			customLogic: props.customLogic,
+		};
+	}
+
+	static getTriggerType(i, trigger) {
+		if (trigger === 'custom') return String(i + 1);
+		if (i > 0) {
+			if (trigger === 'all') return 'AND';
+			if (trigger === 'any') return 'OR';
+		}
+		return '';
+	}
+
+	updateData(i, val, type) {
+		const { conditions } = this.state;
+		if (type === 'value') conditions[i].value = val;
+		else conditions[i][type] = val.selection[0].id;
+		this.setState({ conditions });
+	}
+
+	updateSubData(i, j, val, type) {
+		const { conditions } = this.state;
+		if (type === 'value') conditions[i].conditions[j].value = val;
+		else conditions[i].conditions[j][type] = val.selection[0].id;
+		this.setState({ conditions });
+	}
+
+	updateFormula(data, type) {
+		const { conditions } = this.state;
+		conditions[type] = data;
+		this.setState({ conditions });
+	}
+
+	addCondition() {
+		const { conditions } = this.state;
+		const newCondition = {
+			isGroup: false,
+			resource: '',
+			operator: '',
+			value: '',
+		};
+		conditions.push(newCondition);
+		this.setState({ conditions });
+	}
+
+	addSubCondition(i) {
+		const { conditions } = this.state;
+		const newCondition = {
+			resource: '',
+			operator: '',
+			value: '',
+		};
+		conditions[i].conditions.push(newCondition);
+		this.setState({ conditions });
+	}
+
+	deleteCondition(i) {
+		const { conditions } = this.state;
+		if (conditions.length > 1) {
+			conditions.splice(i, 1);
+			this.setState({ conditions });
+		} else {
+			const newCondition = {
+				resource: '',
+				operator: '',
+				value: '',
+			};
+			this.setState({ conditions: [newCondition] });
+		}
+	}
+
+	deleteSubCondition(i, j) {
+		const { conditions } = this.state;
+		if (conditions[i].conditions.length > 1) {
+			conditions[i].conditions.splice(j, 1);
+			this.setState({ conditions });
+		} else {
+			conditions.splice(i, 1);
+			this.setState({ conditions });
+		}
+	}
+
+	addGroup() {
+		if (!this.props.isChild) {
+			const { conditions } = this.state;
+			const newCondition = {
+				resource: '',
+				operator: '',
+				value: '',
+			};
+			const newGroup = {
+				isGroup: true,
+				triggerType: 'all',
+				conditions: [newCondition],
+			};
+			conditions.push(newGroup);
+			this.setState({ conditions });
+		}
+	}
+
+	updateGroupData(i, val, type) {
+		const { conditions } = this.state;
+		conditions[i][type] = val;
+		this.setState({ conditions });
+	}
+
+	render() {
+		return (
+			<IconSettings iconPath="/assets/icons">
+				<Expression
+					id="expression-example"
+					events={{
+						onChangeTrigger: (val) => {
```

Comment:
Since you've changed the data being passed, the parameters here need to be `(event, data)` with the log function afterwards passing both. Also, the state setter should say `this.setState({ triggerType: data.triggerType });`. See the `resource-selected.jsx` example if needed.